src/components/PeopleLoader/PeopleLoader.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  useState,
+  useCallback,
+  useEffect,
+} from 'react';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+
+const findParent = (
+  people: Person[],
+  parentName: string | null,
+): Person | undefined => {
+  return people.find(parent => parent.name === parentName);
+};
+
+export const PeopleLoader = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const loadPeople = useCallback(async () => {
+    setIsLoading(true);
+
+    try {
+      const peopleFromServer = await getPeople();
+
+      const preparedPeople = peopleFromServer.map(person => ({
+        ...person,
+        mother: findParent(peopleFromServer, person.motherName),
+        father: findParent(peopleFromServer, person.fatherName),
+      }));
+
+      setPeople(preparedPeople);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    loadPeople();
+  }, [loadPeople]);
+
+  if (isError) {
```

Comment:
you can move it to the return, in html code

Reply:
I like it in this case, because if I move to the return I show Error in this style:
![image](https://github.com/mate-academy/react_people-table-basics/assets/124714400/715ba95a-867a-4f69-916f-ebffd7dcb4d8)